src/factory/mastercopyDeployer.ts

```diff
@@ -27,10 +27,10 @@ export const deployMastercopy = async (
 ): Promise<string> => {
   const deploymentTx = mastercopyContractFactory.getDeployTransaction(...args);
 
-  if (Array.isArray(deploymentTx.data) && deploymentTx.data.length > 0) {
-    return await deployMastercopyWithInitData(hre, deploymentTx.data, salt);
+  if (deploymentTx.data == null || deploymentTx.data.length === 0) {
+    throw new Error("Unable to create the deployment data (no init code).");
```

Comment:
Optional: keep original condition: !deploymentTx.data, which is anyway stricter than current